stack_data/formatting.py

```diff
@@ -166,12 +177,13 @@ def format_line(self, line: Line) -> str:
             else:
                 result = " " * len(self.current_line_indicator)
             result += " "
+        else:
```

Comment:
I think this should be removed, the original line `result = result or " "` is enough. A test without `current_line_indicator` but with `show_linenos` would confirm.

Reply:
You are, of course, correct. I had a brain freeze when I added this.

Reply:
I've tested this by creating two additional tests (with or without blank lines included). Do you want these extra tests added in a commit?

Reply:
If you've already written the test etc. then might as well. But no need for extra effort. Just remove these two lines, unless the tests show otherwise.

Reply:
With the two lines, the result is as follows:
```
Traceback (most recent call last):
 File "formatter_example.py", line 85, in blank_lines
     79 | def blank_lines():
     80 |     a = [1, 2, 3]
     82 |     length = len(a)
     85 |     return a[length]
                     ^^^^^^^^^
IndexError: list index out of range
```
Without them, the indentation is slightly less
```
Traceback (most recent call last):
 File "formatter_example.py", line 85, in blank_lines
  79 | def blank_lines():
  80 |     a = [1, 2, 3]
  82 |     length = len(a)
  85 |     return a[length]
                  ^^^^^^^^^
IndexError: list index out of range
```
Which one do you prefer?  Once decided, it probably would make sense to include the tests.

Reply:
Oh, it looks nicer with the extra spaces, I didn't expect that. Especially if the line numbers reach 4 digits. So let's go with the former option, and then I think `result = result or " "` isn't needed.

Reply:
I needed to modified the code when using `blank_lines.HIDDEN` with no current line marker so that it would be consistent with the others. I added three tests, one for each setting.